test/storage/data_table_test.cpp

```diff
@@ -321,4 +347,226 @@ TEST_F(DataTableTests, InsertNoWrap) {
     delete txn;
   }
 }
+
+// Test that insertion into a block does not wrap around even in the presence of deleted slots. This makes compaction
+// a lot easier to write.
+// NOLINTNEXTLINE
+TEST_F(DataTableTests, SimpleNumaTest) {
+  const uint32_t num_iterations = 3;
+  const uint16_t max_columns = 20;
+  for (uint32_t iteration = 0; iteration < num_iterations; ++iteration) {
+    RandomDataTableTestObject tested(&block_store_, max_columns, null_ratio_(generator_), &generator_);
+    // make sure we test the edge case where a block is filled
+    uint32_t num_inserts = iteration == 0
+                               ? tested.Layout().NumSlots()
+                               : std::uniform_int_distribution<uint32_t>(1, tested.Layout().NumSlots())(generator_);
+
+    // Populate the table with random tuples
+    for (uint32_t i = 0; i < num_inserts; ++i)
+      tested.InsertRandomTuple(transaction::timestamp_t(0), &generator_, &buffer_pool_);
+
+    std::vector<storage::col_id_t> all_cols = StorageTestUtil::ProjectionListAllColumns(tested.Layout());
+    EXPECT_NE((!all_cols[all_cols.size() - 1]), -1);
+    storage::ProjectedColumnsInitializer initializer(tested.Layout(), all_cols, num_inserts);
+    auto *buffer = common::AllocationUtil::AllocateAligned(initializer.ProjectedColumnsSize());
+    storage::ProjectedColumns *columns = initializer.Initialize(buffer);
+
+    std::vector<storage::numa_region_t> numa_regions;
+    tested.GetTable().GetNUMARegions(&numa_regions);
+    EXPECT_GE(numa_regions.size(), 1);
+
+#ifdef __APPLE__
+    EXPECT_EQ(numa_regions.size(), 1);
+    EXPECT_EQ(numa_regions[0], storage::UNSUPPORTED_NUMA_REGION);
+#else
+    bool single_numa_system =
+        numa_available() != -1 && numa_regions.size() == 1 && numa_regions[0] == storage::UNSUPPORTED_NUMA_REGION;
+#endif
+
+    for (storage::numa_region_t numa_region : numa_regions) {
+      tested.NUMAScan(transaction::timestamp_t(1), columns, &initializer, &buffer_pool_);
+      for (auto it = tested.GetTable().begin(numa_region); it != tested.GetTable().end(numa_region); ++it) {
+        EXPECT_NE((*it).GetBlock(), nullptr);
+        EXPECT_EQ((*it).GetBlock()->numa_region_, numa_region);
+#ifndef __APPLE__
+        if (numa_available() != -1) {
+          int status;
+          auto *page = static_cast<void *>((*it).GetBlock());
+          if (move_pages(0, 1, &page, nullptr, &status, 0) != -1) {
+            EXPECT_EQ(static_cast<int>(static_cast<int16_t>(numa_region)), status);
+          } else {
+            EXPECT_TRUE(single_numa_system);
+          }
+        }
+#endif
+      }
+    }
+    EXPECT_EQ(num_inserts, columns->NumTuples());
+
+    delete[] buffer;
+  }
+}
+
+TEST_F(DataTableTests, ConcurrentNumaTest) {
```

Comment:
Is it possible to add a test where the out_buffer size is smaller than the blocks stored in each numa_region to see whether current implementation works for that case?

Reply:
We no longer need that function so it has been deleted